src/win/main_win.cc

```diff
@@ -92,17 +97,27 @@ namespace
 
     if (!info[2].IsString())
     {
-      Napi::TypeError::New(env, "A string was expected for the second argument, but wasn't received.").ThrowAsJavaScriptException();
+      Napi::TypeError::New(env, "A string was expected for the third argument, but wasn't received.").ThrowAsJavaScriptException();
       return env.Undefined();
     }
 
     auto id = Utils::utf8ToWideChar(info[0].As<Napi::String>());
     auto title = Utils::utf8ToWideChar(info[1].As<Napi::String>());
     auto body = Utils::utf8ToWideChar(info[2].As<Napi::String>());
+    auto userInfo = L"";
+
+    if (info[3].IsObject())
+    {
+      auto userInfoObject = info[3].As<Napi::Object>();
+      auto userInfoString = Utils::JSONStringify(env, userInfoObject);
+      userInfo = Utils::utf8ToWideChar(userInfoString);
```

Comment:
This is using JavaScript's `JSON.stringify` function to serialize the `userInfo` object, which will be later stored in the `launchArgs` strings, to then be deserialized with JavaScript's `JSON.parse` function before sending it back with the notification callback.

Not very happy about this, but I couldn't think of a better way other than exposing all the Windows notifications API we need to JS and do most of the work from there, but it was a ton of work and didn't want to make this PR bigger than it already is 😂  I might tackle it in a different PR.